frontend/src/pages/NewRunV2.tsx

```diff
@@ -91,16 +93,24 @@ type NewRunV2Props = RunV2Props & PageProps;
 export type SpecParameters = { [key: string]: ComponentInputsSpec_ParameterSpec };
 export type RuntimeParameters = { [key: string]: any };
 
-function hasVersionID(apiRun: ApiRunDetail | undefined): boolean {
-  if (!apiRun) {
+function hasVersionID(
+  apiRun: ApiRunDetail | undefined,
+  apiRecurringRun: ApiJob | undefined,
+): boolean {
+  if (!apiRun && !apiRecurringRun) {
     return true;
   }
   let hasVersionType: boolean = false;
-  if (apiRun.run?.resource_references) {
+  if (apiRun && apiRun.run?.resource_references) {
     apiRun.run.resource_references.forEach(value => {
       hasVersionType = hasVersionType || value.key?.type === ApiResourceType.PIPELINEVERSION;
     });
   }
+  if (apiRecurringRun && apiRecurringRun.resource_references) {
```

Comment:
If `apiRecurringRun` exists, can we clean up any resource_references set by `apiRun`? Otherwise resource_references are mixed and conflicted.

Reply:
If `apiRecurringRun` exists, `apiRun` will be empty.